service/src/test/java/bio/terra/catalog/service/DatasetServiceTest.java

```diff
@@ -190,11 +192,44 @@ void getDatasetPreviewTables() {
                                         .datatype(TableDataType.INTEGER)
                                         .name("column a"))))));
     DatasetPreviewTablesResponse results =
-        datasetService.getDatasetPreviewTables(user, tdrDataset.id());
-    // Test that all the data conversion works as expected
+        datasetService.listDatasetPreviewTables(user, tdrDataset.id());
     assertThat(results, isA(DatasetPreviewTablesResponse.class));
     assertThat(results.getTables().size(), is(1));
     assertThat(results.getTables().get(0), isA(TableMetadata.class));
     assertThat(results.getTables().get(0).isHasData(), is(true));
   }
+
+  @Test
+  void getDatasetPreviewTable() {
+    var tdrDataset =
+        new Dataset(dataset.id(), sourceId, StorageSystem.TERRA_DATA_REPO, metadata, null);
+    var tableName = "tableName";
+    when(datasetDao.retrieve(datasetId)).thenReturn(tdrDataset);
+    when(datarepoService.getPreviewTables(user, tdrDataset.storageSourceId()))
+        .thenReturn(
+            new SnapshotModel()
+                .tables(
+                    List.of(
+                        new TableModel()
+                            .name(tableName)
+                            .rowCount(1)
+                            .columns(
+                                List.of(
+                                    new ColumnModel()
+                                        .datatype(TableDataType.INTEGER)
+                                        .name("column a"))))));
+    when(datarepoService.getPreviewTable(user, tdrDataset.storageSourceId(), tableName))
+        .thenReturn(new SnapshotPreviewModel().result(List.of()));
+    DatasetPreviewTable datasetPreviewTable =
+        datasetService.getDatasetPreview(user, tdrDataset.id(), tableName);
+    assertThat(datasetPreviewTable.getRows().size(), is(0));
+    assertThat(datasetPreviewTable.getColumns().size(), is(1));
```

Comment:
There are specific asserts for collection size
```suggestion
    assertThat(datasetPreviewTable.getRows(), empty());
    assertThat(datasetPreviewTable.getColumns(), hasSize(1));
```